tests/vmi_lifecycle_test.go

```diff
@@ -573,6 +575,55 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 
 				tests.WaitForSuccessfulVMIStart(newVMI)
 			})
+
+			It("[test_id:5275]VMIs with Bridge Networking shouldn't fail after the kubelet restarts", func() {
+				if currentConfiguration.NetworkConfiguration == nil {
+					currentConfiguration.NetworkConfiguration = &v1.NetworkConfiguration{}
+				}
+
+				currentConfiguration.NetworkConfiguration.PermitBridgeInterfaceOnPodNetwork = pointer.BoolPtr(true)
+				kv := tests.UpdateKubeVirtConfigValueAndWait(currentConfiguration)
+				currentConfiguration = kv.Spec.Configuration
+
+				By("starting a VMI on a node")
+				vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+				Expect(err).To(BeNil(), "Should submit VMI successfully")
+
+				// Start a VirtualMachineInstance with bridged networking
+				nodeName := tests.WaitForSuccessfulVMIStart(vmi)
+
+				By("restarting kubelet")
+				pod := renderPkillAllPod("kubelet")
+				pod.Spec.NodeName = nodeName
+				_, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(pod)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("checking if we see the device plugin restart in the logs")
+				virtHandlerPod, err := kubecli.NewVirtHandlerClient(virtClient).Namespace(flags.KubeVirtInstallNamespace).ForNode(nodeName).Pod()
+				Expect(err).ToNot(HaveOccurred(), "Should get virthandler client for node")
```

Comment:
here's a thought.

I don't think you necessarily have to wait for the device plugin restart in virt-handler. Instead how about this.

- start a VMI with bridge network 
- Create the kill pod and wait for the kill pod to succeed (meaning the kubelet's dead)
- start a VMI on the node you killed and wait for that VMI to come online (tells us kubelet is back online)
- then verify the original VMI started in the first step is still healthy.

My thought  process here is killing the kubelet then starting a VMI after the kubelet is down indicates to us the kubelet has cycled. At that point if the original VMI is still online and kicking, then perhaps the CNI check passed?

Reply:
Good news is I think we've confirmed the kill works :).  As long as those checks are tolerant of a few timeouts, I think that should work.  I'll give it a try.